crates/wasm-mutate/src/mutators/codemotion/mutators/if_complement.rs

```diff
@@ -0,0 +1,125 @@
+        then: &[usize],
+        alternative: &Option<Vec<usize>>,
+        newfunc: &mut wasm_encoder::Function,
+        operators: &Vec<crate::mutators::OperatorAndByteOffset>,
+        input_wasm: &'a [u8],
+        ty: &wasmparser::TypeOrFuncType,
+    ) -> crate::Result<()> {
+        // negate the value on the stack
+        newfunc.instruction(&Instruction::I32Eqz);
+        newfunc.instruction(&Instruction::If(map_block_type(*ty)?));
+
+        // Swap, write alternative first
+        if let Some(alternative) = alternative {
+            for ch in alternative {
+                self.write(ast, *ch, newfunc, operators, input_wasm)?;
+            }
+        } else {
+            // Write an unreachable instruction
+            newfunc.instruction(&Instruction::Unreachable);
+        }
+        newfunc.instruction(&Instruction::Else);
+        for ch in then {
+            self.write(ast, *ch, newfunc, operators, input_wasm)?;
+        }
+        newfunc.instruction(&Instruction::End);
+
+        Ok(())
+    }
+}
+
+impl ASTWriter for IfComplementWriter {
+    /// Replaces the default implementation of the if_else_writing
+    fn write_if_else<'a>(
+        &self,
+        ast: &crate::mutators::codemotion::ir::AST,
+        nodeidx: usize,
+        then: &[usize],
+        alternative: &Option<Vec<usize>>,
+        newfunc: &mut wasm_encoder::Function,
+        operators: &Vec<crate::mutators::OperatorAndByteOffset>,
+        input_wasm: &'a [u8],
+        ty: &wasmparser::TypeOrFuncType,
+    ) -> crate::Result<()> {
+        if self.if_to_mutate == nodeidx {
+            self.write_complement(
+                ast,
+                nodeidx,
+                then,
+                alternative,
+                newfunc,
+                operators,
+                input_wasm,
+                ty,
+            )?;
+        } else {
+            self.write_if_else_default(
+                ast,
+                nodeidx,
+                then,
+                alternative,
+                newfunc,
+                operators,
+                input_wasm,
+                ty,
+            )?;
+        }
+        Ok(())
+    }
+}
+
+impl ASTMutator for IfComplementMutator {
+    fn can_mutate<'a>(
+        &self,
+        _: &'a crate::WasmMutate,
+        _: &crate::ModuleInfo,
+        ast: &crate::mutators::codemotion::ir::AST,
+    ) -> bool {
+        ast.has_if()
+    }
+
+    fn mutate_inplace<'a>(
+        &self,
+        _: &'a crate::WasmMutate,
+        _: &crate::ModuleInfo,
+        rnd: &mut rand::prelude::SmallRng,
+        ast: &crate::mutators::codemotion::ir::AST,
+        newfunc: &mut wasm_encoder::Function,
+        operators: &Vec<OperatorAndByteOffset>,
+        input_wasm: &'a [u8],
+    ) -> crate::Result<()> {
+        // Select the if index
+        let if_index = ast
+            .get_ifs()
+            .choose(rnd)
+            .expect("This mutator should check first if the AST contains more than one if");
```

Comment:
I think this is supposed to be

```suggestion
            .expect("This mutator should check first if the AST contains at least one if");
```

?

Am I understanding correctly? This doesn't need multiple `if`s, right?